Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,576 @@
+        :param std_lower_bound: clipping value for the standard deviation, necessary when using very small target variances
+        :param kl_threshold: threshold for the KL-divergence until which std_lower_bound is enforced
+        :param optimize_mean: whether the mean should be changed or considered fixed
+        :param optimize_cov: whether the (co-)variance should be changed or considered fixed
+        :param max_subrtn_retries: how often a failed (median performance < 30 % of performance_lower_bound)
+                                   training attempt of the subroutine should be reattempted
+        """
+        if not isinstance(subroutine, Algorithm):
+            raise pyrado.TypeErr(
+                given=subroutine,
+                expected_type=Algorithm,
+                msg="Subroutine must inherit from *Algorithm* and ExposedSampler!",
+            )
+        if not isinstance(subroutine, ExposedSampler):
+            raise pyrado.TypeErr(
+                given=subroutine,
+                expected_type=ExposedSampler,
+                msg="Subroutine must inherit from Algorithm and *ExposedSampler*!",
+            )
+        if not typed_env(env, DomainRandWrapper):
+            raise pyrado.TypeErr(given=env, expected_type=DomainRandWrapper)
+
+        # Call Algorithm's constructor with the subroutine's properties
+        super().__init__(subroutine.save_dir, max_iter, subroutine.policy, subroutine.logger)
+
+        # Using a Union here is not really correct, but it makes PyCharm's type hinting work
+        # suggest properties from both Algorithm and ExposedSampler.
+        self._subroutine: Union[Algorithm, ExposedSampler] = subroutine
+        self._subroutine.save_name = self._subroutine.name
+
+        self._env = env
+
+        # Properties for the variance bound and kl constraint
+        self._kl_constraints_ub = kl_constraints_ub
+        self._std_lower_bound = std_lower_bound
+        self._kl_threshold = kl_threshold
+
+        # Properties of the performance constraint
+        self._performance_lower_bound = performance_lower_bound
+        self._performance_lower_bound_reached = False
+
+        self._optimize_mean = optimize_mean
+        self._optimize_cov = optimize_cov
+
+        self._max_subrtn_retries = max_subrtn_retries
+
+        self._spl_parameters = [
+            param for param in env.randomizer.domain_params if isinstance(param, SelfPacedDomainParam)
+        ]
+
+    @property
+    def sub_algorithm(self) -> Algorithm:
+        """ Get the policy optimization subroutine. """
+        return self._subroutine
+
+    @property
+    def sample_count(self) -> int:
+        """The amount of samples collected by the subalgorithm."""
+        return self._subroutine.sample_count
+
+    def train(self, snapshot_mode: str = "latest", seed: int = None, meta_info: dict = None):
```

Comment:
I think, this method can be deleted.

Reply:
Done in d39f3add7462870b222d1349f8e07dd71a5b0320